components/button.tsx

```diff
@@ -1,8 +1,8 @@
 import React from "react";
 import styles from "./button.module.scss";
 
-const Button = ({ children, variant, ...rest }) => {
-  return <button className={` ${styles['default']} ${styles[variant]}`} {...rest}>{children}</button>
+const Button = ({ children, variant, disabled = false, ...rest }) => {
+  return <button className={` ${styles['default']} ${styles[variant]} ${disabled ? styles['disabled'] : ''}`} {...rest} disabled={disabled}>{children}</button>
```

Comment:
maybe in the future, we can use https://www.npmjs.com/package/classnames to simplify conditional styles. but this is fine for now, you don't have to change it.